integration/simulation/transaction_injector.go

```diff
@@ -234,13 +266,7 @@ func (m *TransactionInjector) issueInvalidL2Txs() {
 		for fromWallet.Address().Hex() == toWallet.Address().Hex() {
 			toWallet = m.rndObsWallet()
 		}
-		var tx types.TxData
-		switch rand.Intn(2) { //nolint:gosec
-		case 0:
-			tx = NewL2Withdrawal(fromWallet.Address(), obscurocommon.RndBtw(1, 100))
-		case 1:
-			tx = NewL2Transfer(fromWallet.Address(), toWallet.Address(), obscurocommon.RndBtw(1, 500))
-		}
+		tx := NewCustomObscuroWithdrawalTx(obscurocommon.RndBtw(1, 100))
 
```

Comment:
Why only check for invalidly signed withdrawal Txs ? 
Should the transfer txs not be checked too ?

Reply:
The reason it wasn't done initially, is that we don't check in validateChain that those transactions were not included. 
The withdrawals are implicitly checked because the amount of money that is withdrawn here is not counted.

Reply:
That's why I find this confusing.
We're doing invalid signature tests and then relying on a completely different test ( the validateChain amounts) to assure the test is correct.

I'll catalog this as a pet peeve of mine and I'll do a PR when I get a bit a time :)